core/lxdprofile/validate.go

```diff
@@ -63,3 +67,44 @@ func NotEmpty(profiler LXDProfiler) bool {
 	}
 	return false
 }
+
+func NewLXDCharmProfiler(profile Profile) LXDProfiler {
+	return LXDProfiles{Profile: profile}
+}
+
+type LXDProfiles struct {
+	Profile Profile
+}
+
+func (p LXDProfiles) LXDProfile() LXDProfile {
+	return p.Profile
+}
+
+type Profile struct {
+	Config      map[string]string
+	Description string
+	Devices     map[string]map[string]string
+}
+
+func (p Profile) Empty() bool {
+	return len(p.Devices) < 1 && len(p.Config) < 1
+}
+
+func (p Profile) ValidateConfigDevices() error {
```

Comment:
I'm not sure about this, because now we need to edit in two places - charm.v6 and this. If we have no other approach, can we make a note here about where else this exists.

Reply:
added a note